runtime/compiler/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -563,6 +563,23 @@ J9::AheadOfTimeCompile::initializeCommonAOTRelocationHeader(TR::IteratedExternal
          }
          break;
 
+      case TR_BlockFrequency:
+         {
+         TR_RelocationRecordBlockFrequency *blockFrequencyRecord = reinterpret_cast<TR_RelocationRecordBlockFrequency *>(reloRecord);
+         TR_PersistentProfileInfo *profileInfo = TR::comp()->getRecompilationInfo()->getProfileInfo();
+
+         TR_ASSERT(NULL != profileInfo, "PersistentProfileInfo not found when creating relocation record for block frequency\n");
```

Comment:
Probably best to also just fail the compile at this point, there was a new exception recently added `J9::AOTRelocationRecordGenerationFailure` that you could use.